drivers/storage/portworx/component/security.go

```diff
@@ -0,0 +1,212 @@
+package component
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+
+	"github.com/hashicorp/go-version"
+	pxutil "github.com/libopenstorage/operator/drivers/storage/portworx/util"
+	corev1alpha1 "github.com/libopenstorage/operator/pkg/apis/core/v1alpha1"
+	k8sutil "github.com/libopenstorage/operator/pkg/util/k8s"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// SecurityComponentName is the name for registering this component
+	SecurityComponentName = "Security"
+	// SecurityPXPrivateKeysSecretName is the admin secret name for PX security
+	SecurityPXPrivateKeysSecretName = "px-private-keys"
+	// SecuritySharedSecretKey is the data key for any secret containing an auth secret
+	SecuritySharedSecretKey = "shared-secret"
```

Comment:
Just wondering whether we should use `PORTWORX_AUTH_SYSTEM_KEY` and `PORTWORX_AUTH_JWT_SHAREDSECRET` as the keys even in the secret. That way there are not too many names for the same thing. 
What do you think?

Reply:
@piyush-nimbalkar I was following the keys in px-docs example here:
https://docs.portworx.com/cloud-references/security/kubernetes/shared-secret-model-multitenancy/generating-tokens/

I think these keys are fine, since the user won't really be pulling these often. Having lowercase keys inside of the k8s secret also matches the `auth-token` key we have.